Flaky test: test_validate_chat_update_user_input (chat message text truncated)

`tests/playwright/shiny/components/chat/update_user_input/test_chat_update_user_input.py::test_validate_chat_update_user_input[chromium]` failed on PR #2336 (unrelated changes: workflow YAML + docs only) and passed on rerun of the same job.

Failed job: https://github.com/posit-dev/py-shiny/actions/runs/28957774520/job/85921174253 (`playwright-shiny (3.14, chromium, 2)`)

Error:

```
AssertionError: Locator expected to have text 'You said: This chat was sent on behalf of the user. Input will still be focused.'
Actual value: You said: This chat was sent on behalf of the user.
Call log:
  - Expect "to_have_text" with timeout 5000ms
  - waiting for locator("#chat").locator(".shiny-chat-messages-content").locator("> :last-child")
    2 × locator resolved to <div class="shiny-chat-message">…</div>
      - unexpected value ""
    12 × locator resolved to <div class="shiny-chat-message">…</div>
       - unexpected value "You said: This chat was sent on behalf of the user."
```

The assertion timed out while the message text was still missing the trailing " Input will still be focused." — looks like the assertion races the second half of the streamed/updated message.

--- a/tests/playwright/shiny/components/chat/update_user_input/app.py
+++ b/tests/playwright/shiny/components/chat/update_user_input/app.py
@@ -1,3 +1,5 @@
+from itertools import count
+
 from shiny import reactive
 from shiny.express import input, ui
 
@@ -49,6 +51,13 @@ def do_submit_and_focus():
     )
 
 
+# Number each response so the test can tell consecutive submissions of the
+# same message apart (otherwise an assertion on the latest message can be
+# satisfied by the previous, identical response while the new one is still in
+# flight).
+submit_counter = count(1)
+
+
 @chat.on_user_submit
 async def on_user_submit(message: str):
-    await chat.append_message(f"You said: {message}")
+    await chat.append_message(f"You said ({next(submit_counter)}): {message}")

--- a/tests/playwright/shiny/components/chat/update_user_input/test_chat_update_user_input.py
+++ b/tests/playwright/shiny/components/chat/update_user_input/test_chat_update_user_input.py
@@ -40,7 +40,9 @@ def test_validate_chat_update_user_input(page: Page, local_app: ShinyAppProc) ->
     submit.loc.focus()
     submit.click()
     chat.expect_user_input(set_msg)  # User doesn't lose what they had written
-    chat.expect_latest_message("You said: This chat was sent on behalf of the user.")
+    chat.expect_latest_message(
+        "You said (1): This chat was sent on behalf of the user."
+    )
     expect(chat.loc_input_button).to_be_enabled()
     expect(chat.loc_input).not_to_be_focused()
 
@@ -54,7 +56,14 @@ def test_validate_chat_update_user_input(page: Page, local_app: ShinyAppProc) ->
     submit.loc.focus()
     submit.click()
     chat.expect_user_input("")
-    chat.expect_latest_message("You said: This chat was sent on behalf of the user.")
+    # The "(2)" suffix distinguishes this response from the identical message
+    # submitted above. Waiting on it guarantees this submission's round trip
+    # has finished (and the chat input is re-enabled) before the next click;
+    # otherwise the next submit would be silently dropped while the input is
+    # disabled awaiting this response (#2337).
+    chat.expect_latest_message(
+        "You said (2): This chat was sent on behalf of the user."
+    )
     expect(chat.loc_input_button).to_be_disabled()
     expect(chat.loc_input).not_to_be_focused()
 
@@ -63,7 +72,7 @@ def test_validate_chat_update_user_input(page: Page, local_app: ShinyAppProc) ->
     submit_and_focus.click()
     chat.expect_user_input("")
     chat.expect_latest_message(
-        "You said: This chat was sent on behalf of the user. Input will still be focused."
+        "You said (3): This chat was sent on behalf of the user. Input will still be focused."
     )
     expect(chat.loc_input_button).to_be_disabled()
     expect(chat.loc_input).to_be_focused()

